### PR TITLE
47 comment entity

### PR DESCRIPTION
**Pull request by jimmeryn**

### CODE REVIEW

File: src/lib/comment/comment.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  id!: bigint;
+
+  @Column({ name: "creator_id", select: false })
+  creatorId!: bigint;
+
+  @JoinColumn({ name: "creator_id" })
+  @ManyToOne(
+    () => Customer,
+    creator => creator.comments,
+    {}
```
Comment (kaaboaye) [inline src/lib/comment/comment.ts:26]: this empty map is redundant, it would be better to remove it

File: src/web/comments.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  listComments,
+  createComment,
+  getComment
+} from "../lib/comment/comments-service";
+import { BigIntValidation } from "../helpers/bigint-validation";
+
+export const commentRoutes: readonly ServerRoute[] = [
+  {
+    method: "get",
+    path: "/comments",
```
Comment (kaaboaye) [inline src/web/comments.ts:14]: I think we would prefer fetching comments for only one incident instead of all comments

Comment (kaaboaye) [reply src/web/comments.ts:14]: `/incidents/{incidentId}/comments or something like that

File: src/web/comments.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      description: "Creates comment",
+      validate: {
+        payload: Joi.object()
+          .keys({
+            content: Joi.string().required()
+          })
+          .label("CreateCommentInput")
+      }
+    },
+    handler: ({ payload }) => createComment(payload as any)
```
Comment (kaaboaye) [inline src/web/comments.ts:47]: this should also accept `incidentId` and `customerId`

File: src/lib/comment/comment.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  creator!: Customer;
+
+  @Column({ name: "incident_id", select: false })
+  incidentId!: bigint;
+
+  @JoinColumn({ name: "incident_id" })
+  @ManyToOne(
+    () => Incident,
+    incident => incident.comments,
+    {}
```
Comment (kaaboaye) [inline src/lib/comment/comment.ts:33]: {}

File: src/lib/comment/comments-service.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+import { Comment } from "./comment";
+import { transform } from "../../helpers/transform";
+
+function repo(): Repository<Comment> {
+  return getRepository(Comment);
+}
+
+export async function listComments(incidentId: bigint): Promise<Comment[]> {
+  const comments = await repo().find({
+    incidentId: JSON.parse(incidentId.toString())
```
Comment (kaaboaye) [inline src/lib/comment/comments-service.ts:12]: To nie zadziała dobrze bo przerobi inta na floata

Tutaj chyba trzeba będzie scastować przez any :(

**Comment by wojtek717** [comment]
@kaaboaye Resolve conflicts and merge

**Comment by kaaboaye** [comment]
> @kaaboaye Resolve conflicts and merge

@wojtek717  that's not my PR

**Pull request merged**